projects/lexical_shared/lib/lexical/document/store.ex

```diff
@@ -240,24 +269,27 @@ defmodule Lexical.Document.Store do
     GenServer.call(name(), {:close, uri})
   end
 
-  @spec get_and_update(Lexical.uri(), updater()) :: {:ok, Document.t()} | {:error, any()}
+  @spec get_and_update(Lexical.uri(), updater) :: {:ok, Document.t()} | {:error, any()}
   def get_and_update(uri, update_fn) do
     GenServer.call(name(), {:get_and_update, uri, update_fn})
   end
 
-  @spec update(Lexical.uri(), updater()) :: :ok | {:error, any()}
+  @spec update(Lexical.uri(), updater) :: :ok | {:error, any()}
   def update(uri, update_fn) do
     GenServer.call(name(), {:update, uri, update_fn})
   end
 
-  def start_link(_) do
-    GenServer.start_link(__MODULE__, [], name: name())
+  @spec start_link(start_opts) :: GenServer.on_start()
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts, name: name())
   end
 
-  def init(_) do
-    {:ok, State.new()}
+  @impl GenServer
+  def init(opts) do
+    {:ok, State.new(opts)}
```

Comment:
consider ensuring the derivation functions are single arity functions and returning an error if they aren't.